### ISSUE TITLE
TSV axis date format is inconsistent

### ISSUE THREAD

**Issue by SRGDamia1** [ODM2/ODM2DataSharingPortal#564]
Depending on the zoom level, the date is formatted differently.  It should be more consistent.  At all zoom levels, the date should be displayed with numbers (ie, always "04/21/2021", never "Apr-21".  The date should also always appear at least once on the plot, even at the tightest zoom level.

Last year:
![image](https://user-images.githubusercontent.com/5374350/159021621-e2875711-f37b-4e2e-bc47-c94d4c9dc0eb.png)

Last month:
![image](https://user-images.githubusercontent.com/5374350/159021737-3189edb1-0d3d-476c-8a30-6bf51d14154c.png)

Tighter zooms:
![image](https://user-images.githubusercontent.com/5374350/159021930-0d1a29de-aa9d-4224-a82c-01dcad1e9408.png)

< 1 day, date is not displayed at all:
![image](https://user-images.githubusercontent.com/5374350/159022054-8b7601a7-1829-43a2-bc03-05a6c21e509c.png)

**Comment by SRGDamia1** [comment]
Part of #551

**Comment by aufdenkampe** [comment]
@SRGDamia1, I agree that having at least one date at the lowest zoom level would be helpful. However, I also see these as relatively low priority enhancements, given our continued urgent need to do some major code refactoring to address tech debt.

@ptomasula, how easy is it to change this behavior?

**Comment by dbressler75** [comment]
Also, no year is available on x-axis display or when hovering - problematic for general inspection of data,
![hover_noyear](https://user-images.githubusercontent.com/39309124/159354091-7bdf173e-610a-4831-94d8-de7e76d09f0b.JPG)
 etc.

**Comment by ptomasula** [comment]
@aufdenkampe @SRGDamia1 
This data formatting was implemented in commit (20ed2b7)[https://github.com/ODM2/ODM2DataSharingPortal/commit/20ed2b7e9b821737146b14f26b651423379d2dba] and is now live on the staging server. 

However, fully specifying the 'MM-DD-YYYY' for lower zoom levels (where only a few days or even hours are displayed) can lead to the labels overlapping which makes the plot less useful. 

I'm looking for feedback on how best to resolve that. 
A few initial approaches could include:

1. Return to default formatting, we now have the full datestring in the cursor hover pop-up so that will help increase clarity.
2. Increase the spacing between tickmarks (would also reduce the number of tick marks). This would allow the full labels to display, but might also look sparse (particularly at detailed zoom levels) 
3. Further adjust the tick label formatting, but aim to shorten the characters displayed.

**Comment by SRGDamia1** [comment]
Isn't the spacing between ticks dynamic?

**Comment by ptomasula** [comment]
> Isn't the spacing between ticks dynamic?

Correct that tick marks are dynamically generated, but we have control over the scaling factor that is used.

**Comment by SRGDamia1** [comment]
Ok.  "Not overlapping" is most important - if the text overlaps it looks like an error no matter what.

For now, with the year properly in the hover, it would be better to go back to the default ticks that will not overlap and to put finer tuning them as a low priority item.

**Comment by ptomasula** [comment]
Thanks @SRGDamia1, I restored the default tick marks on [staging](https://staging.monitormywatershed.org/). I'll leave this issue open incase we want to revisit x-tick label, but I'll move it over to a future milestone.
